drivers/i2c/i2c_ll_stm32.meta.jinja2

```diff
@@ -0,0 +1,36 @@
+{# Copyright (c) 2018 Nordic Semiconductor ASA #}
+{# SPDX-License-Identifier: Apache-2.0         #}
+
+{% extends 'device_declare.jinja2' %}
+
+{% block init_params %}
+        {% do params.update({'irq_flag':'CONFIG_I2C_STM32_INTERRUPT'}) %}
+        {% do params.update({'config_struct':'i2c_stm32_config'}) %}
+        {% do params.update({'data_struct':'i2c_stm32_data'}) %}
+        {% do params.update({'api_struct':'api_funcs'}) %}
+        {% do params.update({'init_function':'i2c_stm32_init'}) %}
+{% endblock %}
+
+{# Because these blocks are "scoped" all variables in the macro from the
+   base template can be accessed in these blocks #}
+{% block config_struct_body %}
+        .i2c = (I2C_TypeDef *){{ "%x"|format(device['reg']['0']['address']) }},
+        .pclken = {
+                .enr = {{ device['clocks']['0']['bits'] }},
+                .bus = {{ device['clocks']['0']['bus'] }},
+        },
+#ifdef {{ get_param(params, 'irq_flag') }}
+        .irq_config_func = {{ irq_config_function_name }},
+#endif
+        .bitrate = {{ device['clock-frequency'] }},
+{% endblock %}
+
+{% block irq_config_function_body %}
```

Comment:
You should be able to get rid of this block.
IRQ code generation sequence should be able to detect if nodes has irqs and then build ad-hoc function based on irq data extracted from EDTS.
Only input would be isr function prefix: stm32_i2c_isr.
Eventually, prefix might be replaced by "<compat>_isr"

Reply:
Yes, sure, but I wanted to make it possible for the developer to decide what the ISR functions was called and not auto-generate because the template don't have control on the ISR generation.

Reply:
We'll.. Since code generation concept is new in Zephyr, we're trying to get first to a most easy to use and constrained version, which should be used as is by majority of developers.

This should help in adoption by developers and users and avoid having heterogeneous versions of code fragmentation in various zephyr drivers. Once merged, I think we don't want to spend time in reviews arguing with developers that they should not do this and this in the code generation.

This is why a constrained version which support vast majority of cases but limits the individual freedom of each developer in this area is required, from my point of view, at least in its first iteration.

Reply:
Ok, I have removed the irq_config_function_body block and the IRQ_CONNECT and irq_enable lines are now auto generated as in the codegen version.